Add syntax for server management (#1607)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

a|
New sub-clause for `ALTER DATABASE`, which allows modifying the number of servers hosting a database in cluster environments.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
ENABLE SERVER ...
----
a|
New Cypher command for enabling servers.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----

> ALTER SERVER SET OPTIONS

Reviewer: ```suggestion
ALTER SERVER ... SET OPTIONS ...
```

----
a|
New Cypher command for setting options for a server.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
RENAME SERVER ... TO ...
----
a|
New Cypher command for changing the name of a server.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
REALLOCATE DATABASES
----
a|
New Cypher command for re-balancing what servers host which databases.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----

> DEALLOCATE DATABASES FROM SERVERS ...

Reviewer 2: Is it always _SERVERS_ (in plural)?

Author: no, but do we have to list the complete syntax in additions?

Reviewer 2: Would it make sense to write

`DEALLOCATE DATABASES FROM SERVER[S]` like in the CIP?

----
a|

> New Cypher command for moving all databases from servers.

Reviewer: ```suggestion
New Cypher command for removing all databases from servers.
```

Author: not sure I agree, since removing doesn't imply that a different server will start hosting it

Reviewer: true, still not sure I like the original sentence but not sure how to improve it either

Reviewer 2: I'd vote for _moving_ as it does imply that the databases have a destination. The command itself _DEALLOCATE_ sort of implies that as well. I suppose it could be fleshed out a bit, but for brevity's sake, I'd say keep it as it is.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
DROP SERVER ...
----
a|

> New Cypher command for dropping servers.

Reviewer: Add one row for the new privileges as well (similar to impersonate/alias in 4.4 but maybe have all privileges in one clump and not one row for each?)

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
ALTER SERVER ... SET OPTIONS ...
----
a|
New Cypher command for setting options for a server.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
RENAME SERVER ... TO ...
----
a|
New Cypher command for changing the name of a server.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
REALLOCATE DATABASES
----
a|
New Cypher command for re-balancing what servers host which databases.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
DEALLOCATE DATABASE[S] FROM SERVER[S] ...
----
a|
New Cypher command for moving all databases from servers.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
DROP SERVER ...
----
a|
New Cypher command for dropping servers.

a|
label:functionality[]
label:added[]
[source, cypher, role="noheader"]
----
SHOW SERVERS
----
a|
New Cypher command for listing servers.

a|
label:functionality[]
label:new[] +
New privileges:
[source, cypher, role="noheader"]
----
SERVER MANAGEMENT
----
[source, cypher, role="noheader"]
----
SHOW SERVERS
----
a|

> New privilege that allows a user to create, modify, reallocate, deallocate, drop and list servers.

Reviewer: ```suggestion
New privileges that allow a user to create, modify, reallocate, deallocate, drop and list servers.
```
